2024/json-ld-wg/index.html

```diff
@@ -178,39 +179,62 @@ <h2>Motivation and Background</h2>
+          of JSON-LD related formats which orbit around the JSON-LD processing model, and specifying improved
+          context retrieval methods for greater clarity and stability.
         </p>
+
       </div>
 
       <section id="scope" class="scope">
         <h2>Scope</h2>
 				<p>
-					The Working Group will extend the recommendations defining the JSON-LD specifications (i.e., <a href="https://www.w3.org/TR/json-ld11/">JSON-LD 1.1</a>, <a href="https://www.w3.org/TR/json-ld11-api/">JSON-LD 1.1 API</a>, <a href="https://www.w3.org/TR/json-ld11-framing/">JSON-LD 1.1 Framing</a>)
-          with the features introduced by <a href="https://www.w3.org/TR/rdf12-concepts">RDF 1.2</a>
-          in addition to open Errata and requested features
-          as noted on the groups <a href="https://github.com/orgs/w3c/projects/84">Project Management Page</a>.
-          Together, these specifications provide a JSON format for Linked Data,
-          enabling interoperation in a way which is familiar to and usable by software engineers.
-					<!-- Note that the JSON-LD APIs are not browser specific; while appropriate for use within browsers, they are not limited to such use, and there is no requirement for browsers to implement them natively. -->
-				</p>
+					The Working Group will evolve the recommendations defining the JSON-LD specifications 
+          (i.e., <a href="https://www.w3.org/TR/json-ld11/">JSON-LD 1.1</a>, 
+          <a href="https://www.w3.org/TR/json-ld11-api/">JSON-LD 1.1 API</a>, and
+          <a href="https://www.w3.org/TR/json-ld11-framing/">JSON-LD 1.1 Framing</a>) by handling Errata 
+          and by adding requested features as recorded on the group's 
+          <a href="https://github.com/orgs/w3c/projects/84">Project Management Page</a>. In order to handle these, 
+          the Working Group will issue new versions of these recommendations, possibly with new features,
+          in order to render future usage and evolutions easier. In particular, the Working Group will handle 
+          two important security related problems:
+        </p>
+
+
+        <ol>
+          <li>[Placeholder: safe mode]</li>
+          <li>[Placeholder: context file hash, and related stuff]</li>
+        </ol>
+
+        <!-- Note that the JSON-LD APIs are not browser specific; while appropriate for use within browsers, they are not limited to such use, and there is no requirement for browsers to implement them natively. -->
+
+        <p>
+          The development of the 1.2 versions of the JSON-LD specification will follow the principle of
+          backward compatibility. This means that any valid JSON-LD 1.1 document will remain valid for 
+          JSON-LD 1.2, and the RDF graphs generated by the 1.1 and 1.2 versions of the JSON-LD API 
+          specifications, will be <a href="https://www.w3.org/TR/rdf11-concepts/#graph-isomorphism">isomorphic</a>.
```

Comment:
A few remarks on this:

* some documents will cause "safe mode" to signal an error or warning (where JSON-LD 1.1 processors would silently ignore parts of the JSON not covered by the context) → JSON-LD 1.2 will not procude a different graphs, but depending on how "hard" the safe mode is, it may err instead of producing any graph. I think that it is still ok with the current text, but just wanted to make sure we are on the same page.

* this text focuses on the `toRdf` function of the API, but shouldn't we commit that the other functions (`compact`, `expand`...) also return the same result?

* of course, this commitment to return the same result does not hold in the cases where we have an errata identifying or inconsistency a bug in JSON-LD 1.1 right? Should we spell it out explicitly?

Given the above, maybe we want to change this to:
```suggestion
          JSON-LD 1.2, and that the algorithms of the JSON-LD 1.2 API will yield the same results as those of the JSON-LD 1.1 API,
          –barring any bug or inconsistency of the JSON-LD 1.1 specifications identified in the errata.
```
```

Reply:
An 1.2 processor may follow a slightly different logic when generating, say, bnode id-s. We _know_ that the generated graphs are "the same" but not everyone in the AC will realize that. That is why I rely on the formal notion of isomorphism in the sentence...

(+1 to the reference to inconsistency problem)

Reply:
Tiny fix to @pchampin's suggestion.
```suggestion
          JSON-LD 1.2, and that the algorithms of the JSON-LD 1.2 API will yield the same results as those of the JSON-LD 1.1 API
          — barring any bug or inconsistency of the JSON-LD 1.1 specifications identified in the errata.
```

Reply:
(Initial tiny fix was to change `-` to `— `. Second tiny fix was to delete the `,` preceding the `—`)

Reply:
Pending on whether we just say "same" or refer to the graph isomorphism. @pchampin ?

Reply:
I actually looked at the text, and there is a way to "merge" both versions. The new paragraph could be:

>    Development of the 1.2 versions of the JSON-LD specifications will follow the principle of backward compatibility. This means that any valid JSON-LD 1.1 document will remain valid for JSON-LD 1.2, and that the algorithms of the JSON-LD 1.2 API will yield <a href="https://www.w3.org/TR/rdf11-concepts/#graph-isomorphism">isomorphic</a> results to those of the JSON-LD 1.1 API — barring any bug or inconsistency of the JSON-LD 1.1 specifications identified in the errata.

Would that work for everyone?

Reply:
That works for me, thanks.

Reply:
Thanks, have update the text/